tensorflow_datasets/structured/HTRU2.py

```diff
@@ -0,0 +1,118 @@
+Here the legitimate pulsar examples are a minority positive class, and spurious examples the majority negative class. 
+At present multi-class labels are unavailable, given the costs associated with data annotation.
+The data set shared here contains 16,259 spurious examples caused by RFI/noise, and 1,639 real pulsar examples. 
+These examples have all been checked by human annotators.
+"""
+
+_URL = "http://archive.ics.uci.edu/ml/machine-learning-databases/00372/HTRU2.zip"
+
+
+class Htru2(tfds.core.GeneratorBasedBuilder):
+  """Dataset for Predicting a Pulsar Star"""
+
+  VERSION = tfds.core.Version('0.1.0',
+                              experiments={tfds.core.Experiment.S3: False})
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict({
+          "Mean of the integrated profile" : tf.float64,
+          "Standard deviation of the integrated profile" : tf.float64,
+          "Excess kurtosis of the integrated profile" : tf.float64,
+          "Skewness of the integrated profile" : tf.float64,
+          "Mean of the DM-SNR curve" : tf.float64,
+          "Standard deviation of the DM-SNR curve" : tf.float64,
+          "Excess kurtosis of the DM-SNR curve" : tf.float64,
+          "Skewness of the DM-SNR curve" : tf.float64,
+          "Class" : tf.float32 # 0 for noise, 1 for pulsar
+        }),
+        supervised_keys=None,
+        homepage="https://archive.ics.uci.edu/ml/datasets/HTRU2",
+        citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    """Returns SplitGenerators."""
+    path = dl_manager.download_and_extract(_URL)
+
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            num_shards=1,
+            gen_kwargs={
+                'file_path': path,
+            }),
+    ]
+
+  def _generate_examples(self, file_path):
+    """Yields examples."""
+    with tf.io.gfile.GFile(file_path + "/HTRU_2.csv", "r") as csvfile:
+      features = [
+          "Mean of the integrated profile",
+          "Standard deviation of the integrated profile",
+          "Excess kurtosis of the integrated profile",
+          "Skewness of the integrated profile",
+          "Mean of the DM-SNR curve",
+          "Standard deviation of the DM-SNR curve",
+          "Excess kurtosis of the DM-SNR curve",
+          "Skewness of the DM-SNR curve",
+          "Class" # 0 for noise, 1 for pulsar
+      ]
+      for i in csvfile.readlines():
+        lst = i.split(",")
+        dictionary = {}
+        for j in range(len(lst) - 1):
+          if j % len(features) < len(features) - 1:
+            dictionary[features[j % len(features)]] = float(lst[j])
+            
+          elif j % len(features) == len(features) - 1:
+            dictionary[features[j % len(features)]] = float(lst[j])
+            yield dictionary
```

Comment:
This is an odd way of yielding, do the tests even pass? You also need to yield a unique id in addition to the dictionary of features